crypto/sr25519.go

```diff
@@ -37,6 +38,25 @@ func NewSr25519Keypair(priv *sr25519.SecretKey) (*Sr25519Keypair, error) {
 	}, nil
 }
 
+// NewSr25519KeypairFromSeed returns a new Sr25519Keypair given a seed
+func NewSr25519KeypairFromSeed(seed []byte) (*Sr25519Keypair, error) {
+	buf := [32]byte{}
+	// ignore last 6 bytes otherwise the seed might be too large
```

Comment:
Hmm... creating a keypair without using the entire seed doesn't seem right. Can you explain?

Reply:
this goes down to the underlying schnorrkel library, basically if the number is too large to fit on the curve, it'll cause an error. there is a function I need from the ristretto library that can reduce it but it's not exported, still trying to find a fix for this

Reply:
Should we add a `TODO` and issue for this?

Reply:
nvm, it has been fixed